### ISSUE TITLE
Feature request: Adding custom cosmology

### ISSUE THREAD

**Issue by steveschulze** [nikhil-sarin/redback#147]
Hi,

Redshift might not be a good distance measurement for galaxies outside the Hubble flow. It would be good to add a dummy cosmology class that mimics the behaviour of astropy.cosmology.

Suggestion:

```
from astropy import units as u

class user_cosmology():
    
    def __init__(self, dl=0):
        self.dl = 0

    def set_luminosity_distance(cls, dl):
        cls.dl  = dl
    
    def luminosity_distance(self, redshift):
        return self.dl

dummy = user_cosmology()
dummy.set_luminosity_distance(40*u.kpc)

```

When the routine is called (where 2 is an arbitrary redshift):

```
dummy.luminosity_distance(2)
```

the object returns

```
40 kpc 
```

like the astropy.cosmology class

Cheers,

Steve

**Comment by nikhil-sarin** [comment]
I'll address this in a pull request in the future unless you'd like to do this. It should be fairly straightforward and you might be able to get to it faster than I can. 

Currently, the dl is hardcoded as 
`cosmo.luminosity_distance(redshift)`
in every model. Where cosmo is the astropy Planck18 cosmology.

The modification required would be to make this cosmo an optional keyword argument via 
`cosmology = kwargs.get('cosmology', cosmo)`. Then have dl be 
`dl = cosmology.luminosity_distance(redshift).cgs.value`

Where users can pass their own dummy class like the above or any other astropy cosmology. If a user does not pass one in then Planck18 is used.

**Comment by nikhil-sarin** [comment]
Implemented in #164
